lib/ai/tools/web-search.ts

```
@@ -64,54 +81,83 @@ export async function executeWebSearchTool({
+      (async () => {
+        const { fullStream } = streamText({
+          model: perplexity(selectedModel),
+          system: systemPrompt,
+          messages: toVercelChatMessages(messages),
+          maxTokens: 2048,
+          abortSignal,
+          onFinish: async ({ finishReason }: { finishReason: string }) => {
+            if (supabase) {
+              await handleChatWithMetadata({
+                supabase,
+                chatMetadata: config.chatMetadata!,
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)